File: artifacts/captionai-mobile/targets/main-app/TokenSync.m
#import <React/RCTBridgeModule.h>

@interface RCT_EXTERN_MODULE(TokenSync, NSObject)
RCT_EXTERN_METHOD(setToken:(NSString *)token)
RCT_EXTERN_METHOD(clearToken)
RCT_EXTERN_METHOD(setNiche:(NSString *)niche)
@end

File: artifacts/captionai-mobile/targets/main-app/TokenSync.swift
import Foundation

@objc(TokenSync)
class TokenSync: NSObject {
  private static let appGroup = "group.com.captionai.app"
  private static let tokenKey  = "captly_auth_token"
  private static let nicheKey  = "captly_user_niche"

  @objc static func requiresMainQueueSetup() -> Bool { return false }

  @objc func setToken(_ token: String) {
    UserDefaults(suiteName: Self.appGroup)?.set(token, forKey: Self.tokenKey)
  }

  @objc func clearToken() {
    UserDefaults(suiteName: Self.appGroup)?.removeObject(forKey: Self.tokenKey)
  }

  @objc func setNiche(_ niche: String) {
    UserDefaults(suiteName: Self.appGroup)?.set(niche, forKey: Self.nicheKey)
  }
}
